src/App.tsx

```diff
@@ -15,36 +16,121 @@ export const goodsFromServer = [
+  const preparedGoods = [...goods];
+
+  if (sortGood !== SortType.NONE) {
+    preparedGoods.sort((good1, good2) => {
+      let result: number;
+
+      switch (sortGood) {
+        case SortType.ALPHABET:
+          result = String(good1).localeCompare(String(good2));
+          break;
+
+        case SortType.LENGTH:
+          result = String(good1).length - String(good2).length;
+          break;
+
+        default:
+          result = 0;
+          break;
+      }
+
+      return result;
+    });
+  }
+
+  if (isReversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortGood, setSortGood] = useState<SortType>(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const preparedGoods = getPreparedGoods({
+    goods: goodsFromServer,
+    sortGood,
+    isReversed,
+  });
+
+  const isInitialOrder = sortGood === SortType.NONE && !isReversed;
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          onClick={() => {
+            setSortGood(SortType.ALPHABET);
+          }}
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortGood !== SortType.ALPHABET,
+          })}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          onClick={() => {
+            setSortGood(SortType.LENGTH);
+          }}
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortGood !== SortType.LENGTH,
+          })}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {!isInitialOrder && (
+          <button
+            onClick={() => {
+              setSortGood(SortType.NONE);
+              setIsReversed(false);
+            }}
+            type="button"
+            className="button is-danger is-light"
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {preparedGoods.map(good => (
+          <li key={good} data-cy="Good">
```

Comment:
Using the item itself as the `key` in a list can lead to issues if the items are not unique. Consider using a unique identifier for each item if possible, such as an index or a unique property of the item.